tests/e2e/loadbalancer.go

```diff
@@ -737,3 +883,87 @@ func inClusterTestReachableHTTP(cs clientset.Interface, namespace, nodeName, tar
 
 	return nil
 }
+
+// Gather information from the cluster to help debug failures.
+// - Resource events
+// - All namespace events
+// - Cloud controller manager logs
+// - Service status
+func gatherResourceEvents(ctx context.Context, cs clientset.Interface, namespace, resourceName string) {
```

Comment:
Also isolated https://github.com/kubernetes/cloud-provider-aws/pull/1217
to enhance the review here.